devices/Ft6xx6x/samples/Program.cs

```diff
@@ -1,55 +1,226 @@
+                const int x0R = 271;
+                const int y0 = 263;
+
+                TouchEventArgs touchEventArgs = new TouchEventArgs();
+
+                //WARNING: Next assigment should be EventCategory.Touch but Touch Category not yet defined on main/develop
+                touchEventArgs.eventCategory = EventCategory.Unknown;
+                touchEventArgs.X1 = x;
+                touchEventArgs.Y1 = y;
+
+                //r^2 in pixels^2
+                int r2 = (int)((rMm * pxPerMm) * (rMm * pxPerMm));
+
+                int tmpL = 0;
+                int tmpM = 0;
+                int tmpR = 0;
+                int tmpY = 0;
+
+                tmpY = (y - y0) * (y - y0);
+                tmpL = (x - x0L) * (x - x0L);
+                tmpL += tmpY;
+                tmpM = (x - x0M) * (x - x0M);
+                tmpM += tmpY;
+                tmpR = (x - x0R) * (x - x0R);
+                tmpR += tmpY;
+
+                if (tmpL <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int)TouchEventSubcategory.LeftButton;
+                }
+                else if (tmpM <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.MiddleButton;
+                }
+                else if (tmpR <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.RightButton;
+                }
+                else
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.SingleTouch;
+                }
+
+                TouchPanelProcessor.OnTouchedEvent(touchEventArgs.eventSubCategory, touchEventArgs.X1, touchEventArgs.Y1);
+
+            }
+
+            // Browse our samples repository: https://github.com/nanoframework/samples
+            // Check our documentation online: https://docs.nanoframework.net/
+            // Join our lively Discord community: https://discord.gg/gCyBu8T
+        }
+
+    }
+    public class TouchEventArgs : EventArgs
+    {
+        public EventCategory eventCategory { get; set; }
+        public int eventSubCategory { get; set; }
+        public int X1 { get; set; }
+        public int Y1 { get; set; }
+        public int X2 { get; set; }
+        public int Y2 { get; set; }
+
+        //public DateTime TimeReached { get; set; }
+    }
+    public enum TouchEventSubcategory
     {
-        var point = sensor.GetPoint(true);
-        Debug.WriteLine($"ID: {point.TouchId}, X: {point.X}, Y: {point.Y}, Weight: {point.Weigth}, Misc: {point.Miscelaneous}, Event: {point.Event}");
+        Unknown = 1000,
+        LeftButton = 1001,
+        MiddleButton = 1002,
+        RightButton = 1003,
+        SingleTouch = 1004,
+        DoubleTouch = 1005
+    }
+
+    public delegate void TouchEventHandler(object sender, TouchEventArgs e);
+
+    public class TouchPanelProcessor
+    {
+        public static event TouchEventHandler TouchedEvent;
```

Comment:
and why all static?

Reply:
Bear with me, i just learned NET events / event handling 48h ago..i following this YouTube video https://www.youtube.com/watch?v=841BAFuoXGE ...there he explicitly does that on his Processor class stating that he "doesn't want (or need) anyone to instantiate objects from that class.... 
![image](https://user-images.githubusercontent.com/13390283/147744724-23d161bc-20f6-4ed8-8a47-7885f35b75c9.png)
...and actually , this brings a very good point, in my previous life as C/C++ developer of embedded systems (elevator controller) it was basically forbidden to dynamically instantiate objects i.e. using "new" (most objects were Singletons created at startup by the Factory pattern) .....so, my question here is regarding exactly that, do you have anything on the coding guidelines that talk about that ? i.e. suggestion to avoid dynamic instantiation (and therefore automatic destruction)? ....coming back to your question, in any given Core2 .NET application the TouchPanelProcessor would be a Singleton if not an static class (or a class with static member functions) , so no need at all to allow instantiating more than one object of it.... What do you suggest ?